qrexec-lib/pack.c

```diff
@@ -107,6 +107,18 @@ void wait_for_result(void)
             case EINVAL:
                 call_error_handler("File copy: Corrupted data from packer%s\"%s\"", last_filename_prefix, last_filename);
                 break;
+            case EILSEQ:
+                // FIXME: the sanitized filename is not useful here, but the unsanitized filename
+                // is dangerous (which is likely why it was rejected in the first place!).
+                call_error_handler("Forbidden character in file or link target");
```

Comment:
Still, some hint which name it is would be useful. Sanitized name will be distorted a bit (say that in the message?), but still can help to find the issue.